service/matching/task_queue_partition_manager.go

```diff
@@ -348,12 +372,17 @@ func (pm *taskQueuePartitionManagerImpl) unloadFromEngine() {
 	pm.engine.unloadTaskQueuePartition(pm)
 }
 
+// Pass either versionSet or build ID
 func (pm *taskQueuePartitionManagerImpl) getVersionedQueue(
 	ctx context.Context,
-	version string,
+	versionSet string,
+	buildId string,
 	create bool,
 ) (physicalTaskQueueManager, error) {
-	tqm, err := pm.getVersionedQueueNoWait(version, create)
+	if pm.partition.Kind() == enumspb.TASK_QUEUE_KIND_STICKY {
+		panic("versioned queues can't be used in sticky partitions")
```

Comment:
do we really need to panic? I know it's a bug but let's return a serviceerror.Internal instead?